### PR TITLE
[BugFix] NPE when abort txn on HiveTable

### PR DESCRIPTION
**Pull request by padmejin**
## What type of PR is this：
- [x] bugfix
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Fix this NPE.
```
2022-08-31 15:23:16,838 WARN (thrift-server-pool-413|24710) [FrontendServiceImpl.loadTxnRollback():1138] catch unknown result.
java.lang.NullPointerException: null
        at com.starrocks.transaction.DatabaseTransactionMgr.abortTransaction(DatabaseTransactionMgr.java:1300) ~[starrocks-fe.jar:?]
        at com.starrocks.transaction.DatabaseTransactionMgr.abortTransaction(DatabaseTransactionMgr.java:1238) ~[starrocks-fe.jar:?]
        at com.starrocks.transaction.GlobalTransactionMgr.abortTransaction(GlobalTransactionMgr.java:424) ~[starrocks-fe.jar:?]
        at com.starrocks.service.FrontendServiceImpl.loadTxnRollbackImpl(FrontendServiceImpl.java:1165) ~[starrocks-fe.jar:?]
        at com.starrocks.service.FrontendServiceImpl.loadTxnRollback(FrontendServiceImpl.java:1128) [starrocks-fe.jar:?]
        at com.starrocks.thrift.FrontendService$Processor$loadTxnRollback.getResult(FrontendService.java:2306) [starrocks-fe.jar:?]
        at com.starrocks.thrift.FrontendService$Processor$loadTxnRollback.getResult(FrontendService.java:2286) [starrocks-fe.jar:?]
        at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:38) [libthrift-0.13.0.jar:0.13.0]
        at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:38) [libthrift-0.13.0.jar:0.13.0]
        at com.starrocks.common.SRTThreadPoolServer$WorkerProcess.run(SRTThreadPoolServer.java:310) [starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]

```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|3|3|100.00%|[]|

**Comment by padmejin** [comment]
run starrocks_admit_test

**Review by chaoyli — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java [hunk 1]
```diff
@@ -1243,7 +1243,10 @@ public void abortTransaction(long transactionId, boolean abortPrepared, String r
             for (Long tableId : transactionState.getTableIdList()) {
                 Table table = db.getTable(tableId);
                 if (table != null) {
-                    listeners.add(stateListenerFactory.create(this, table));
+                    TransactionStateListener listener = stateListenerFactory.create(this, table);
+                    if (listener != null) {
+                        listeners.add(stateListenerFactory.create(this, table));
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java:1248]:
```suggestion
                        listeners.add(listener);
```

**Comment by padmejin** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Comment by padmejin** [comment]
@Mergifyio backport branch-2.4 branch-2.3
¡Muchas gracias!

**Pull request merged**